R/infrastructure-git.R

```diff
@@ -102,19 +107,65 @@ use_github <- function(auth_token = github_pat(), private = FALSE, pkg = ".",
   }
 
   message("* Creating GitHub repository")
-  create <- github_POST("user/repos", pat = auth_token, body = list(
-    name = jsonlite::unbox(pkg$package),
-    description = jsonlite::unbox(gsub("\n", " ", pkg$title)),
-    private = jsonlite::unbox(private)
-  ))
+  # github_POST() needs a path and a host (protocol and hostname)
+  #
+  # In the default case, the supplied value for host is
+  # "https://api.github.com". For github_POST(), the host
+  # will not change and path will be "user/repos".
+  #
+  # If you are using enterprise github, you may have supplied a host that
+  # looks like "https://github.hostname.com/api/v3". For github_POST(),
+  # the host will be "https://github.hostname.com", and path will be
+  # "api/v3/user/repos".
+  #
+  url <- httr::parse_url(host)
+  path_prefix <- url$path
+  url$path <- ""
+  host <- httr::build_url(url)
+  path <- "user/repos"
+  if (!identical(path_prefix, "")){
+    path <- file.path(path_prefix, path)
+  }
+
+  create <-
+    github_POST(
+      path,
+      pat = auth_token,
+      body = list(
+        name = jsonlite::unbox(pkg$package),
+        description = jsonlite::unbox(gsub("\n", " ", pkg$title)),
+        private = jsonlite::unbox(private)
+      ),
+      host = host
+    )
 
   message("* Adding GitHub remote")
   r <- git2r::repository(pkg$path)
   origin_url <- switch(protocol, https = create$clone_url, ssh = create$ssh_url)
   git2r::remote_add(r, "origin", origin_url)
 
   message("* Adding GitHub links to DESCRIPTION")
-  use_github_links(pkg$path)
+  # use_github_links() expects a host (protocol and hostname)
+  #
+  # In the default case, for use_github_links(), the host is
+  # "https://github.com". However in this function, the default host is
+  # "https://api.github.com" - so we have to make the translation.
+  #
+  # If the hostname is "api.github.com", hostname becomes "github.com";
+  # other hostnames are not changed. The protocol is preserved.
+  #
+  # Any path in the url is removed.
+  #
+  # The resulting url, which should have only a protocol and hostname, is
+  # rebuilt and sent to use_github_links()
+  #
+  url <- httr::parse_url(host)
```

Comment:
And this should be pulled out into a function like `github_url_from_host()` or similar.